src/components/NumberInput/NumberInput.js

```diff
@@ -106,13 +112,19 @@ export default class NumberInput extends Component {
       value: this.state.value,
     };
 
+    let error = null
+    if (invalid) {
+      inputProps['data-invalid'] = true;
+      error = <div className="bx--form-requirement">{invalidText}</div>
+    }
+
     return (
       <div className="bx--form-item">
         <label htmlFor={id} className="bx--label">
           {label}
         </label>
         <div className={numberInputClasses}>
```

Comment:
Based on the way we're using `data-invalid` and a general sibling selector, I think we'll have to put the `data-invalid` on the wrapping `bx--number` `div`. Putting `bx--form-requirement` adjacent to `bx--number` (as opposed to most other inputs that have it adjacent to the actual input) seems to avoid most styling issues. I've got a PR up in the base repo that addresses this.